File: exercismdotio/python/phone-number/phone_number.py
class Phone():
	def __init__(self, rawPhoneNumber):
		# pulling out non-digits without regular expressions:
		cleanedUpPhoneNumber = []
		for character in rawPhoneNumber:
			if character in '1234567890':
				cleanedUpPhoneNumber.append(character)
		cleanedUpPhoneNumber = ''.join(cleanedUpPhoneNumber)

		# check that this is a logical phone number
		if len(cleanedUpPhoneNumber) == 10:
			self.number = cleanedUpPhoneNumber
		elif len(cleanedUpPhoneNumber) == 11 and cleanedUpPhoneNumber[0] == '1':
			self.number = cleanedUpPhoneNumber[1:]
		else:
			self.number = '0000000000'

	def area_code(self):
		return self.number[:3]

	def pretty(self):
		return '(%s) %s-%s' % (self.number[:3], self.number[3:6], self.number[6:])
		

File: exercismdotio/javascript/phone-number/phone-number.js
/*jshint esversion: 6 */

var PhoneNumber = function(rawPhoneNumber) {
    'use strict';

    // removing all non-digits
    rawPhoneNumber = rawPhoneNumber.replace(/[^0-9]/g, '');

    if (rawPhoneNumber.length === 10) {
        this.phoneNumber = rawPhoneNumber;
    } 
    else if (rawPhoneNumber.length === 11 && rawPhoneNumber[0] === '1') {
        this.phoneNumber = rawPhoneNumber.slice(1); // chop off first digit, which is '1'
    }
    else {
        this.phoneNumber = '0000000000'; // bad phone numbers
    }
};


PhoneNumber.prototype.number = function() {
    return this.phoneNumber;
};

PhoneNumber.prototype.areaCode = function() {
    return this.phoneNumber.slice(0, 3);
};

PhoneNumber.prototype.toString = function() {
    return `(${this.areaCode()}) ${this.phoneNumber.slice(3, 6)}-${this.phoneNumber.slice(6,10)}`;
};



module.exports = PhoneNumber;